snowpark_extensions/dataframe_reader_extensions.py

```diff
@@ -72,7 +63,50 @@ def _format(self, file_type: str) -> "DataFrameReader":
             self._file_type = file_type
         else:
             raise Exception(f"Unsupported file format {file_type}")
+
+    def _csv(self,path: Union[str, List[str]],schema: Optional[Union[StructType, str]] = None,sep: Optional[str] = None,encoding: Optional[str] = None,quote: Optional[str] = None,
+             escape: Optional[str] = None,comment: Optional[str] = None,header: Optional[Union[bool, str]] = None,inferSchema: Optional[Union[bool, str]] = None,
+             ignoreLeadingWhiteSpace: Optional[Union[bool, str]] = None,ignoreTrailingWhiteSpace: Optional[Union[bool, str]] = None,nullValue: Optional[str] = None,
+             nanValue: Optional[str] = None,positiveInf: Optional[str] = None,negativeInf: Optional[str] = None,dateFormat: Optional[str] = None,timestampFormat: Optional[str] = None,
+             maxColumns: Optional[Union[int, str]] = None,maxCharsPerColumn: Optional[Union[int, str]] = None,maxMalformedLogPerPartition: Optional[Union[int, str]] = None,
+             mode: Optional[str] = None,columnNameOfCorruptRecord: Optional[str] = None,multiLine: Optional[Union[bool, str]] = None,charToEscapeQuoteEscaping: Optional[str] = None,
+             samplingRatio: Optional[Union[float, str]] = None,enforceSchema: Optional[Union[bool, str]] = None,emptyValue: Optional[str] = None,locale: Optional[str] = None,
+             lineSep: Optional[str] = None,pathGlobFilter: Optional[Union[bool, str]] = None,recursiveFileLookup: Optional[Union[bool, str]] = None,modifiedBefore: Optional[Union[bool, str]] = None,
+             modifiedAfter: Optional[Union[bool, str]] = None,unescapedQuoteHandling: Optional[str] = None) -> "DataFrame":
+        params = {k: v for k, v in locals().items() if v is not None}
+        params.pop("self", None)
+        params.pop("path", None)
+        params.pop("schema", None)
+        if schema:
+            self.schema(schema)
+        files = get_file_paths(path)
+        session = context.get_active_session()  
+        stage = f'{session.get_fully_qualified_current_schema()}.{_generate_prefix("TEMP_STAGE")}'
+        session.sql(f'create TEMPORARY stage if not exists {stage}').show()
+        stage = get_stage(self, session, files, stage)
+        for key, value in params.items():
+          self = self.option(key, value)
+        return self.__csv(f"@{stage}")
     
+    def get_file_paths(path: Union[str, List[str]]):
+        if isinstance(path,list):
+            return path
+        else:
+            return [path]
+        
+    def get_stage(self, session, files: List[str], stage: str):
+        print(f"Uploading files using stage {stage}")
+        for file in files:
+            if file.startswith("file://"): # upload local file
+                session.file.put(file,stage)
+            elif file.startswith("@"): #ignore it is on an stage
+                return self._read_semi_structured_file(file,format)
```

Comment:
this return look a bit suspicious, because was the return at the original code, but here is does not represent the get_stage result, isn't it?